### PR TITLE
User type value

### PR DESCRIPTION
**Pull request by shierei**
This fix is to avoid a getter of a user defined type to return a garbage as discussed in Issue thread #110.

### CODE REVIEW

File: include/ponder/detail/userpropertyimpl.inl [hunk 1]
```diff
@@ -69,7 +69,7 @@ UserPropertyImpl<A>::UserPropertyImpl(IdRef name, A&& accessor)
 template <typename A>
 Value UserPropertyImpl<A>::getValue(const UserObject& object) const
 {
-    return ToUserObject<A::canWrite>::get(
+    return ToUserObject<A::RefTraits::isRef>::get(
```
Comment (billyquith) [inline include/ponder/detail/userpropertyimpl.inl]: Yep, that's a typo. Well done.

**Comment by shierei** [comment]
Sorry about that. How do I run your regression tests?Thanks,-Shie-reiOn Jan 15, 2019 8:26 AM, Bill Quith <notifications@github.com> wrote:Tests are currently failing. I'll have to do more investigation.

**Comment by shierei** [comment]
I saw the failures in pondertest. It is on the test that parses an xml document. The parsing is done recursively by passing a UserObject supposedly with a reference to its C++ instance to collect the parsing result. The first call of ArchiveReader<ARCHIVE>::read() passes a UserObject with a reference to a C++ instance by explicitly calling makeRef(). That is fine for the first call. Subsequent recursive call passes the UserObject constructed with property.get(object).to<UserObject>(). The test expects this second level UserObject property also be constructed with a reference to its C++ instance. With my fix, that is not the case. I have yet to understand why it should be the case.
-Shie-rei
    On Tuesday, January 15, 2019, 8:26:35 AM PST, Bill Quith <notifications@github.com> wrote:  
 
 
Tests are currently failing. I'll have to do more investigation.

**Comment by shierei** [comment]
I believe the bug that this pull request tries to fix caused the existing serialization test to succeed. I have modified the tests and updated the pull request. Now the regression tests pass. Please help review again.

**Comment by shierei** [comment]
Can I commit now?  Thanks.

**Comment by shierei** [comment]
I meant if you could update the branch now. If you want me to do more updates, feel free to let me know. Thanks.

File: test/ponder/serialise.cpp [hunk 2]
```diff
@@ -150,21 +150,23 @@ TEST_CASE("Can serialise using RapidXML")
...
         
         {
-            std::unique_ptr<Ref> r{ new Ref };
-            REQUIRE(r != nullptr);
-            
-            r->m_instance.m_i = 89;
-            r->m_instance.setF(0.75f);
-            r->m_instance.m_s = "stringy";
-
+            std::unique_ptr<Simple> s1{ new Simple };
```
Comment (billyquith) [inline test/ponder/serialise.cpp]: The point of this test is that we save a nested object instance. This has been changed and will just test the same as the previous test.

**Pull request merged**